spec/ddtrace/tracer_spec.rb

```diff
@@ -187,7 +187,6 @@
 
           context 'is a block' do
             it 'yields to the error block and raises the error' do
-              expect_any_instance_of(Datadog::Span).to_not receive(:set_error)
```

Comment:
Turns out a specific combination of `any_instance_of` and JRuby causes issues like this: https://app.circleci.com/pipelines/github/DataDog/dd-trace-rb/1694/workflows/b0a5b0af-711b-4aca-81db-3c99b1d504e5/jobs/81614

I've opened a `rspec-mocks` issue with this report: https://github.com/rspec/rspec-mocks/issues/1338
I wasn't able to confidently fix it, unfortunately 😕

Reply:
Appreciate if you could respond in your ticket. Unfortunately, most probably nobody is going to fix it for you.

Reply:
Hey @pirj, I'll follow up on the ticket, but long story short: I dedicated around 2 days trying to fix it in `rspec-mocks` but I wasn't able to.

Reply:
Thanks a lot. I understand, if it's tolerable on your side, it might not worth the effort. However, if you decide to proceed, I'll be happy to provide any support.